### PR TITLE
RRFS cloud base/top: diagnose using cloud fraction

### LINKED ISSUE #790: Request to modify the cloud-base height diagnostic used in RRFS
RRFS physics developers at GSL would like to modify the RRFS cloud-base height diagnostic to use cloud fraction instead of resolved cloud water and cloud ice.  This modified cloud-base field will then be used for verification statistics at GSL.

To support this request, a few changes are proposed to `CLDRAD.f`, such as:

- Around line 1654, add a block for `if(MODELNAME == 'FV3R')`.

- Then, within this block, determine `IBOTGr` using a cloud-fraction value.  The subsequent calculation of `CLDZ` uses `IBOTGr`.

This issue can be assigned to @jaymes-kenyon.  Copying @EricJames-NOAA for awareness.

### PR DESCRIPTION
**Pull request by jaymes-kenyon**
This code updates the cloud-base and cloud-top height (and pressure) diagnostics for RRFS applications, addressing issue #790.  Rather than using explicit (grid-scale) cloud water and ice, the proposed code uses the cloud fraction provided by the model.  
- For cloud-base height (and pressure), the code searches upward, then locates the model mid-layer point where cloud fraction is >= 2%.  The underlying interface level is then assigned as the cloud base.  
- For cloud-top height (and pressure), the search proceeds downward.   However, the model mid-layer point is retained as the cloud top, rather then the overlying interface level.

This code has been tested in a RRFS case and produces satisfactory results.  Thanks to @joeolson42 (NOAA / GSL / EPAD / Model Physics Branch) for the collaboration.

### CODE REVIEW

File: sorc/ncep_post.fd/CLDRAD.f [hunk 1]
```diff
@@ -1650,6 +1650,7 @@ SUBROUTINE CLDRAD
     !
     !--- Grid-scale cloud occurs when the mixing ratio exceeds QCLDmin
     !    or in the presence of snow when RH>=95% or at/above the PBL top.
+    !    However, for RRFS (FV3R), simply use a threshold cloud fraction.
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CLDRAD.f:1654]: @jaymes-kenyon Can you add a change log with your name at the beginning session of this subroutine?

Comment (jaymes-kenyon) [reply sorc/ncep_post.fd/CLDRAD.f:1654]: Good catch, @WenMeng-NOAA — all set now.

**Comment by WenMeng-NOAA** [comment]
The UPP RT on Hera indicate the following changes:
```
fv3r(rrfs): 
NATLEV--
1614:1346818648:PRES:cloud base:rpn_corr=0.670028:rpn_rms=62502.5
1615:1349115561:HGT:cloud base:rpn_corr=0.365044:rpn_rms=5670.54
1616:1351360814:PRES:grid scale cloud bottom level:rpn_corr=0.671265:rpn_rms=62155.3
1617:1353662250:PRES:cloud top:rpn_corr=0.617659:rpn_rms=54259.6
1618:1355736553:HGT:cloud top:rpn_corr=0.617302:rpn_rms=6871.4
1619:1357903926:TMP:cloud top:rpn_corr=0.598246:rpn_rms=371.684
1620:1359887723:PRES:grid scale cloud top level:rpn_corr=0.617659:rpn_rms=54259.6

PRSLEV--
893:847611164:PRES:cloud base:rpn_corr=0.670028:rpn_rms=62502.5
894:849908077:HGT:cloud base:rpn_corr=0.365044:rpn_rms=5670.54
898:858689763:PRES:grid scale cloud bottom level:rpn_corr=0.671265:rpn_rms=62155.3
899:860991199:PRES:cloud top:rpn_corr=0.617659:rpn_rms=54259.6
900:863065502:HGT:cloud top:rpn_corr=0.617302:rpn_rms=6871.4
901:865232875:TMP:cloud top:rpn_corr=0.598246:rpn_rms=371.684
903:868039118:PRES:grid scale cloud top level:rpn_corr=0.617659:rpn_rms=54259.6

3drtma:
NATLEV--
1614:1299926568:PRES:cloud base:rpn_corr=0.481097:rpn_rms=78085.8
1615:1302511707:HGT:cloud base:rpn_corr=0.331731:rpn_rms=5672.06
1616:1305069244:PRES:grid scale cloud bottom level:rpn_corr=0.482067:rpn_rms=77580.7
1617:1307659749:PRES:cloud top:rpn_corr=0.406446:rpn_rms=69175.5
1618:1310167527:HGT:cloud top:rpn_corr=0.442935:rpn_rms=7134.64
1619:1312722741:TMP:cloud top:rpn_corr=0.438933:rpn_rms=462.415
1620:1315052114:PRES:grid scale cloud top level:rpn_corr=0.406446:rpn_rms=69175.5

PRSLEV--
893:835275641:PRES:cloud base:rpn_corr=0.481097:rpn_rms=78085.8
894:837860780:HGT:cloud base:rpn_corr=0.331731:rpn_rms=5672.06
898:846923435:PRES:grid scale cloud bottom level:rpn_corr=0.482067:rpn_rms=77580.7
899:849513940:PRES:cloud top:rpn_corr=0.406446:rpn_rms=69175.5
900:852021718:HGT:cloud top:rpn_corr=0.442935:rpn_rms=7134.64
901:854576932:TMP:cloud top:rpn_corr=0.438933:rpn_rms=462.415
903:857676827:PRES:grid scale cloud top level:rpn_corr=0.406446:rpn_rms=69175.5

hafs:
HURPRS--
725:784488845:TMP:cloud top:rpn_corr=0.494309:rpn_rms=431.326
```

**Comment by WenMeng-NOAA** [comment]
@jaymes-kenyon I would think all changes from my UPP RT (see above) are expected results. You may access the test case for RRFS on Hera at
/scratch1/NCEPDEV/stmp2/Wen.Meng/fv3r_2023062800. Please let me know if you see issues.

**Comment by WenMeng-NOAA** [comment]
@BinLiu-NOAA and @LinZhu-NOAA There will be changes in cloud top temperature for hafs products with updates of cloud base/top height calculation for all regional FV3 models (See my testing above) in this PR. Please let me know if you have concerns.

**Comment by jaymes-kenyon** [comment]
Thanks, @WenMeng-NOAA — these results would be expected.  Indeed, changes in cloud-top height (via the `ITOP` value) will also produce changes in cloud-top temperature.  This occurs at line 2644 in `CLDRAD.f`.

**Comment by WenMeng-NOAA** [comment]
> Thanks, @WenMeng-NOAA — these results would be expected. Indeed, changes in cloud-top height (via the `ITOP` value) will also produce changes in cloud-top temperature. This occurs at line 2644 in `CLDRAD.f`.

@jaymes-kenyon Thanks for confirming.

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA I completed the UPP RT on Hera at /scratch1/NCEPDEV/stmp2/Wen.Meng. You may copy the new baselines for Hera and start the RT on Orion.

**Comment by FernandoAndrade-NOAA** [comment]
> @FernandoAndrade-NOAA I completed the UPP RT on Hera at /scratch1/NCEPDEV/stmp2/Wen.Meng. You may copy the new baselines for Hera and start the RT on Orion.

Understood thank you. I'll get started on updating Hera BLs. Orion is currently down for maintenance, I'll run RTs when it's available later this afternoon.

**Comment by FernandoAndrade-NOAA** [comment]
Orion RTs have completed with the same changes in results for 3drtma, fv3r, and hafs, baselines will be updated for these tests on Orion. Hera BLs have been updated.

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Pull request merged**